### ISSUE TITLE
Build module can't find msbuild for VS 2017

### ISSUE THREAD

**Issue by jamesashley1** [Microsoft/MixedRealityToolkit-Unity#893]
The underlying process call to cmd.exe is returning the error "'vswhere' is not recognized as an internal or external command". Verified that vswhere.exe is in the installation directory. Call works fine when I open up a command window and paste it in. Just not working from System.Diagnostics.Process .

**Comment by StephenHodgson** [comment]
Did you install in a custom location?

**Comment by StephenHodgson** [comment]
Were you able to figure out exactly what was causing this to fail in the editor and not in cmd.exe?

**Comment by jamesashley1** [comment]
I was not. Investigated for several hours. 

----- Original Message -----

**Comment by RustamIrzaev** [comment]
I faced the same behavior. It works directly from the command line, but not from `BuildDeployTools.cs` script.
As a temporary workaround I did:

1. Execute this script from the command line `cmd.exe "/C cd "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer" && vswhere -version 15.0 -products * -requires Microsoft.Component.MSBuild -property installationPath"`
2. Copy that output (in my case it is `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community`).
3. Go to `BuildDeployTools.cs` to method `CalcMSBuildPath` and insert one `if` statement just after `using (var vswhereP = new System.Diagnostics.Process())` block :

```csharp
if (string.IsNullOrEmpty(output))
    return @"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community" + @"\MSBuild\" + msBuildVersion + @"\Bin\MSBuild.exe";
```


Now it works, but this solution is ugly. Just a workaround...
